azurelinuxagent/ga/exthandlers.py

```diff
@@ -2283,6 +2289,28 @@ def get_resource_limits(self, extension_name, str_version):
 
         return ResourceLimits(self.data.get('resourceLimits', None))
 
+    def report_invalid_boolean_properties(self, ext_name):
+        """
+        Check that the specified keys in the handler manifest has boolean values.
+        """
+        for key in ['reportHeartbeat', 'continueOnUpdateFailure', 'supportsMultipleExtensions']:
+            value = self.data['handlerManifest'].get(key, False)
+            if not isinstance(value, bool):
+                msg =("In the handler manifest: '{0}' has a non-boolean value [{1}] for boolean type. Please change it to a boolean value. "
+                      "For backward compatibility, 'true' (case insensitive) is accepted, and other values default to False.").format(key, value)
```

Comment:
I would not advertise this, since we do not want to encourage its use

Reply:
you mean this part `"For backward compatibility, 'true' (case insensitive) is accepted, and other values default to False.`

Reply:
Yup, we don't want to encourage the use of strings as booleans. We need to supported for extensions that already do it, but we should not let other extensions know that we do.